### DISCUSSION TITLE
Display to controller cable

### DISCUSSION THREAD

**Discussion by dav0000000** [danielnilsson9/bbs-fw#59] [General]
I have just managed to break my cable inside the 5 pin green julet plug on the display side. The pins are all still perfect, it broke inside the connector from pulling it on and off so many times. It results in an error 30 - communication error. If I wiggle it around it will work again. This is not the first time this has happened and I'm honestly not surprised.

My question is, can I rewire to have two female plugs. One for the display and the other for the programming cable. Then with the display turned off, could could I plug in and use the programming cable without issues? Does this work or does the display turn on when the programming cable is plugged in and the motor is turned on?

**Comment by danielnilsson9** [comment]
I have done that to once... Be careful to no short any of the high voltage lines to the low voltage ones.

It's hard to say if that will work, it entirely depends on if the display pulls any of the RX/TX lines in any direction while it's off, if so the communication will not work. The display should not turn on when plugging the programming cable, the power has not been back feed when I have done that at least (DPC-18).

**Comment by dav0000000** [comment]
Thanks so much for your quick reply. This is the 4th time I've fixed the display plug on my bike and I had to do it once on the other bike. I really must be more careful.

This time I shortened the display cable (it works again now), I have to shorten an extension I have now, to run from display to the harness. That way I can just replace the extension next time it happens. I've probably replugged thousands of times.

I did have to be very careful this time as the cable I bought had different colour wires and they were all over the place. I had to connect the black display wire to the red on the new cable. Each wire went to a different color. I was a bit worried when I plugged it in but it was all good.

I have the dpc-18 display too, without the ECO - SPORT. So you are saying you are not sure it will work even with the DPC-18. I think I'll just make the extension and then build a splitter later if I can't find one ready made.

**Comment by danielnilsson9** [comment]
Sound like a plan, my setup is a bit different since I got very tired of unplugging the cable for each test of the firmware. I made a separate board with Arduino as a proxy between the display and the controller to filter out event logs and send them over bluetooth to my phone/computer.

Then I have a switch for resetting the controller to enter flash mode.
![IMG_20220910_195118](https://user-images.githubusercontent.com/5458667/189495833-f405b0ae-0d67-4218-8d1a-c2a78ebe1f29.jpg)

**Comment by zeynelozturk** [comment]
Split display cable also works for me, with 500C screen and programming cable. I cannot turn on the screen when USB cable is connected, and that's it. I also used Speeed app this way, before using this firmware and it worked.

I haven't tried / remember connecting anything while screen was on.

**Reply by dav0000000** [reply]
I got the cables to build this last week so will hopefully get around to it soon. Thanks.

**Comment by stts2** [comment]
Anybody grease up their connectors with dialectric grease? My cables are crazy tight to try to pull apart. I was thinkin of picking up that grease. and gooing up the connectors. Seals corrosion out, and I was hoping that is would make disconnecting a breeze.
